### PR TITLE
Hapi: Pass toolkit to context function

### LINKED ISSUE #1382: Apollo Server 2 Hapi response toolkit is undefined
I'm trying to include a cookie in the response but the response toolkit is undefined when accessed in the context method.

```js
new ApolloServer({
  typeDefs,
  resolvers,
  context: async ({ request, h }) => {
    return { ... };
  },
})
```
https://github.com/apollographql/apollo-server/tree/master/packages/apollo-server-hapi

`h` is undefined

My current server configuration

```js
const server = new ApolloServer({
    typeDefs,
    resolvers,
    schemaDirectives,
    context: async ({ request, h }) => {
      console.log('h', h);
      return {
        request,
          h,
      };
    },
    formatError: (error) => formatError(error, app),
    tracing: true,
  });
```

Am I doing something wrong here?

### PR DESCRIPTION
**Pull request by maxnachlinger**
Fixes #1382

TODO:

* [x] Update CHANGELOG.md with your change (include reference to issue & this PR)
* [x] Make sure all of the significant new logic is covered by tests
* [x] Rebase your changes on master so that they can be merged easily
* [x] Make sure all tests and linter rules pass

### CODE REVIEW

**Comment by maxnachlinger** [comment]
Hey folks, I'd love a review of this micro `, h]` PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ All of the packages in the `apollo-server` repo are released with the same versi
 
 ### vNEXT
 
-- Pass the response tollkit to the context function. [#1407](https://github.com/apollographql/apollo-server/pull/1407)
```
Comment (maxnachlinger) [inline CHANGELOG.md]: Hey my `tollkit` was... you know, trying to generate revenue for this awesome open-source project or something :) Me no make text good.

**Comment by evans** [comment]
@maxnachlinger This looks great! Thank you so much. It would be awesome to have a test for the context function to ensure that there is a toolkit. Adding it [here](https://github.com/apollographql/apollo-server/blob/47dcfdb4605524894d209cefe4efaab4256aba3e/packages/apollo-server-hapi/src/ApolloServer.test.ts#L58-L64) is the best bet. Let me know if I can help with that more!

File: packages/apollo-server-hapi/src/ApolloServer.test.ts [hunk 2]
```diff
@@ -61,6 +61,17 @@ describe('apollo-server-hapi', () => {
       const server = new ApolloServer({ typeDefs, resolvers });
       return server.applyMiddleware({ app });
     });
+
+    it('accepts a context function with an object argument containing a request and response toolkit key', async () => {
+      const app = new Server();
+      const context = async ({ request, h }) => {
+        expect(request, 'request argument should exist').to.exist;
```
Comment (maxnachlinger) [inline packages/apollo-server-hapi/src/ApolloServer.test.ts]: @evans - I'm not sure the assertions below are even run :(

File: packages/apollo-server-hapi/src/ApolloServer.test.ts [hunk 3]
```diff
@@ -223,6 +234,33 @@ describe('apollo-server-hapi', () => {
       await apolloFetch({ query: '{hello}' });
     });
 
+    it('passes each request and response toolkit through to the context function', async () => {
```
Comment (maxnachlinger) [inline packages/apollo-server-hapi/src/ApolloServer.test.ts]: @evans This seems like a better test, what do you think?

**Comment by maxnachlinger** [comment]
@evans This is ready for another review

**Comment by jessicabyrne** [comment]
+1 , we'd like to get this toolkit into our resolvers!

**Comment by evans** [comment]
@maxnachlinger Thank you so much!

**Comment by chasevida** [comment]
Curious as to how this is then used within the resolver. When I tried to modify the response with the toolkit, any headers that are set for example are stripped out in the final response presumably by Apollo. How are people intending this be used?

**Pull request merged**